### PR TITLE
[REFACTOR] Shibboleth SSO Integration

### PR DESCRIPTION
**Pull request by finnlestrange-uofg**
> [!Caution]
> This PR will remove `next-auth` and allow only authentication through a `Shibboleth SSO`.

### Implementation Plan
* `middleware.ts` - catch `HTTP` headers
* Use Next v14 server actions.
* New user session type to store a shibboleth user `{guid, displayName, unscopedAffiliations}`
* Fully remove `next-auth` and refactor `schema.prisma`

### CODE REVIEW

File: src/middleware.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  const id = z.string().parse(shib_guid);
+  const displayName = z.string().parse(shib_displayName);
+  const email = "";
+  const groups = [] as string[];
+
+  const user = await getUserAction(db, {
+    guid: id,
+    displayName,
+    email,
+    groups,
```
Comment (finnlestrange-uofg) [inline src/middleware.ts:28]: Set groups here but not in `schema.prisma`? not added to schema yet?

File: prisma/schema.prisma [hunk 2]
```diff
@@ -323,3 +323,9 @@ model VerificationToken {
 
     @@unique([identifier, token])
 }
+
+model NewUser {
+    id    String @id
+    name  String
+    email String @unique
```
Comment (finnlestrange-uofg) [inline prisma/schema.prisma:330]: Do we need another field to store the shibboleth groups, `unscopedAffiliations`, i.e. staff, student etc. ?

Comment (pkitazos) [reply prisma/schema.prisma:330]: How we store them depends I suppose on whether a user can belong to multiple groups/`unscopedAffiliations` with the same account, i.e. would they use the same logins as a student and staff. 

If a user can belong to multiple and we need to store them we can create a `UserGroups` or `UserAffiliations` table to store this information.

We probably don't need to store them on the user model, as long as we create the appropriate `StudentDetails`, `SupervisorDetails` and `AdminInSpace` rows when a user gets invited to a space.

Comment (finnlestrange-uofg) [reply prisma/schema.prisma:330]: Ahhhh okay, that makes more sense to do it that way.

**Pull request closed without merging**